### ISSUE TITLE
twisted.test.test_dict fails under Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8732]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8732|
|Type|enhancement|
|Created|2016-08-07 22:55:17Z|
|Branch|https://github.com/twisted/twisted/tree/8732-rodrigc-dict-py3|

If I run:

python3 bin/trial twisted.test.test_dict

it fails under Python 3

<details><summary>Searchable metadata</summary>

```
trac-id__8732 8732
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8732_rodrigc_dict_py3 8732-rodrigc-dict-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1470610517607241 1470610517607241
changetime__1470664916467872 1470664916467872
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/475

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_2"></a>|@acabhishek942 commented|
|-|-|

changes looks good.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset 39c506ffcc09ec5ce8764ca8e669e02cdcee1848
```
#!CommitTicketReference repository="" revision="39c506ffcc09ec5ce8764ca8e669e02cdcee1848"
Merge pull request #475 from twisted/8732-rodrigc-dict-py3

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8732

Fix twisted.test.test_dict on Python 3
```
